src/nci_util.c

```diff
@@ -887,10 +1009,16 @@ nci_discovery_ntf_copy_array(
                 ptr += G_ALIGN8(src->param_len);
                 if (src->param) {
                     NciModeParam* dest_param = (NciModeParam*)ptr;
+                    const gsize copied = nci_mode_param_copy_impl(dest_param,
+                        src->param, src->mode);
 
-                    *dest_param = *src->param;
-                    dest->param = dest_param;
-                    ptr += G_ALIGN8(sizeof(*src->param));
+                    if (copied) {
+                        dest->param = dest_param;
+                        ptr += G_ALIGN8(copied);
+                    } else {
+                        GWARN("ModeParam is not NULL but non copyable");
```

Comment:
Hmm... I wonder how this could be possible at all. If NciModeParam isn't NULL then it was parsed. And if it was parsed, then it must be copiable, right?

Reply:
I prefer not to drop this, just to be sure that everything is working as expected